packages/vscode-extension/lib/expo_router_helpers.js

```diff
@@ -6,6 +6,18 @@ export function computeRouteIdentifier(pathname, params) {
   return query ? `${pathname}?${query}` : pathname;
 }
 
+export function compareNavigationDescriptors(a, b) {
+  if (a.pathname !== b.pathname) {
+    return false;
+  }
+  for (const key in a.params) {
+    if (a.params[key] !== b.params[key]) {
+      return false;
+    }
+  }
```

Comment:
NIT:
I'd use `Array.some()` or `Array.every()` for this sort of "for loop checking some predicate", since they generally better communicate the intent, like
```suggestion
  return Object.keys(a.params).every(key => a.params[key] === b.params[key])
```
That said, shouldn't we also check that the keys are the same for both objects?
For example, for:
```
a.params = { a: "a" }
b.params = { a: "a", b: "b" }
```
this method will return `true`